services/management/management.go

```diff
@@ -0,0 +1,59 @@
+package management
+
+import (
+	"github.com/AlekSi/pointer"
+	"github.com/percona/pmm/api/inventorypb"
+	"github.com/percona/pmm/api/managementpb"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+func addNode(tx *reform.TX, addNodeParams *managementpb.AddNodeParams, address string) (*models.Node, error) {
+	var nodeType models.NodeType
+	switch addNodeParams.NodeType {
+	case inventorypb.NodeType_GENERIC_NODE:
+		nodeType = models.GenericNodeType
+	case inventorypb.NodeType_CONTAINER_NODE:
+		nodeType = models.ContainerNodeType
+	case inventorypb.NodeType_REMOTE_NODE:
+		nodeType = models.ContainerNodeType
```

Comment:
I wonder if we have that mapping in some other place. Maybe we can extract it to the `services` root package.

Reply:
Ok, and wrong type on that line.

Reply:
actually we have it in management/node.go, so we can extract it to separate function in this file.